src/common/StepSequence.cpp

```diff
@@ -61,8 +61,11 @@ int StepSequence::getMaxSteps() const
   int numSteps = (mEndPoint - mStartPoint) / mStepSize;
   if (!mIncludeEndpoints)
   {
-    // Return numSteps + 1 for the start
-    return numSteps + 1;
+    if (fabs(mStartPoint + mStepSize * numSteps - mEndPoint) < 1e-7)
```

Comment:
I found the current logic somewhat difficult to follow. Can we do something like:

```
int numSteps = (mEndPoint - mStartPoint) / mStepSize;
if (fabs(mStartPoint + mStepSize * numSteps - mEndPoint) > 1e-7)
  numSteps++; // include start
return numSteps + mIncludeEndpoints;
```